pkg/accounts/kubernetes/validator.go

```diff
@@ -32,21 +36,49 @@ type kubernetesAccountValidator struct {
 }
 
 func (k *kubernetesAccountValidator) Validate(spinSvc v1alpha2.SpinnakerServiceInterface, c client.Client, ctx context.Context, log logr.Logger) error {
-	if err := k.validateSettings(ctx, log); err != nil {
+	err := k.validateSettings(ctx, log)
+	if err != nil {
 		return err
 	}
-	if k.account.Auth != nil && k.account.Auth.UseServiceAccount && spinSvc == nil {
-		// don't validate if the spinnaker account needs a k8s service account to run validations, and there's no SpinnakerService yet
-		return nil
+	if k.account.Auth != nil && k.account.Auth.UseServiceAccount {
```

Comment:
Can we do it below in `makeClientFromServiceAccount`? That seems like a more natural place instead of passing `spinsvc` around.

Reply:
We still need to pass spinSvc around because it's received as an argument when running validations as part of SpinnakerService itself.

Reply:
Okay. That still seems more logical to do it closer to where we need it.